roles/download/tasks/main.yml

```diff
@@ -26,5 +38,9 @@
     - not skip_downloads | default(false)
     - download.enabled
     - item.value.enabled
-    - (not (item.value.container | default(false))) or (item.value.container and download_container)
+    - (not (item.value.container | default(false))) or (item.value.container and download_container and (not (((item.value.repo | regex_replace('^docker\.io/(library/)?', '')) ~ ':' ~ item.value.tag in node_images.stdout.split(',')) or (item.value.repo ~ '@sha256:' ~ (item.value.sha256 | default(None)) in node_images.stdout.split(','))) or download_always_pull))
```

Comment:
I have tried to move the when-condition into the loop, but at the moment I lack the creativity to map this logic in the loop.

Perhaps you have a few more hints for me. Otherwise I have to admit that I can only manage it with the condition.